modules/service/src/main/scala/lucuma/itc/legacy/IOLocalItc.scala

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.itc.legacy
+
+import cats.effect.Async
+import cats.effect.kernel.syntax.all.*
+import cats.syntax.all.*
+import lucuma.itc.UpstreamException
+import lucuma.itc.legacy
+
+/**
+ * Wraps local calls to ITC to ensure fairness
+ */
+case class FLocalItc[F[_]: Async](itcLocal: LocalItc):
+  private val F = Async[F]
+
+  def calculateCharts(jsonParams: String): F[GraphsRemoteResult] =
+    (F.cede *> F.delay(itcLocal.calculateCharts(jsonParams)).guarantee(F.cede)).flatMap {
```

Comment:
Nice 👍

Reply:
Thanks, that part is calling legacy code and it can do IO and some heavy calculations

Reply:
Oh, is it doing actual I/O as well, and not just calculationsl? Then it should probably be in `blocking`.